### PR TITLE
Reduced cognitive complexity in tx_segment.js

### PR DESCRIPTION
**Pull request by mrickard**
## Proposed Release Notes
Reduced cognitive complexity in tx_segment.js

## Links
https://issues.newrelic.com/browse/NEWRELIC-5293

### CODE REVIEW

File: lib/metrics/normalizer/tx_segment.js
[the remark this answers is not in the document]
Comment (mrickard) [reply lib/metrics/normalizer/tx_segment.js:1]: Let me try that. I'm also thinking of refactoring that outer for loop--as it is, this gets us under 15, but it's still a weird mix of iteration styles.

[the remark this answers is not in the document]
Comment (mrickard) [reply lib/metrics/normalizer/tx_segment.js:1]: Easy enough to resolve this one. Still working on the outer for loop.

**Review by jmartin4563 — DISMISSED** [review]

File: lib/metrics/normalizer/tx_segment.js [hunk 1]
```diff
@@ -44,26 +59,10 @@ TxSegmentNormalizer.prototype.normalize = function normalize(path) {
     }
     const fragment = path.slice(prefix.length)
     const parts = fragment.split('/')
-    const result = []
-    let prev
-    let segment
-
-    for (let j = 0; j < parts.length; j++) {
-      segment = parts[j]
+    result = []
```
[the remark this answers is not in the document]
Comment (mrickard) [reply lib/metrics/normalizer/tx_segment.js:62]: My reading of this is that it needs to be reset with each of the terms. I could initialize it in the let, but it's going to need to be reset here.

[the remark this answers is not in the document]
Comment (jordigh) [reply lib/metrics/normalizer/tx_segment.js:62]: Oh, of course, it's inside a loop, my bad.

[the remark this answers is not in the document]
Comment (mrickard) [reply lib/metrics/normalizer/tx_segment.js:62]: It's a tricky one! Trickier as a nested loop, too.

**Pull request merged**